lib/streamlit/DeltaGenerator.py

```diff
@@ -1936,18 +1933,30 @@ def file_uploader(self, element, label, type=['*'], key=None):
 
         Examples
         --------
-        >>> file = st.file_uploader("Upload a image", type=([".png"]))
+        >>> file = st.file_uploader("Upload a image", type=[".png"])
         >>> if file != None:
         >>>     st.image(file)
-
         """
         element.file_uploader.label = label
-        element.file_uploader.type[:] = type
+
+        if type is not None:
+            total_len = sum([len(i) for i in type])
+            if len(type) == total_len: # If all elements are single chars, is a string
+                if type[0] == '.':
+                    element.file_uploader.type[:] = [''.join(type)] # example: type=".csv"
+                else:
+                    element.file_uploader.type[:] = ['.' + ''.join(type)] # example: type="csv"
+            else:
+                element.file_uploader.type[:] = map(lambda x: x if x[0] == '.' else '.' + x , type) # example: type=["png", ".jpeg"]
```

Comment:
I don't understand why this logic is needed. The file type passed by the user should **never** have a period.

So you could replace the whole logic above with just:

```py
if _type(type) in string_types:
  type = [type]

element.file_uploader.type[:] = [
   "." + t for t in type
]
```

Also: this is a bit of a nit-pick, but I think the best place to add a period to the types is actually in JS rather than in Python. Otherwise you're exposing a JS implementation detail to Python for no good reason. So I'd just replace this whole thing with:

```py
if _type(type) in string_types:
  type = [type]

element.file_uploader.type[:] = type
```

If it turns out to be annoying to do this in JS, though, feel free to use the first code snippet above instead.